### ISSUE TITLE
[Bug?] Steam Client power plan profile resetting

### ISSUE THREAD

**Issue by ssj4maiko** [Valkirie/HandheldCompanion#311]
I always set up the Steam Client to either have no power plan, or just to set it to minimum, since it's just navigating for most of the time, there is no need to waste 15W or 18W in it.

It works for some time, and after resetting or some other trigger that I couldn't define yet, I notice the cooler being louder, check it and sure enough it has been reset to 15W or 18W (lately it started giving me 18, I guess it may be copying "the last game I had set up" or something?).

I also tried deleting them the profile, or just disabling it, doesn't really matter.

I haven't properly defined what's the trigger for this bug, it doesn't happen always, and it's relatively easy change it back, but it has been happening more and more.

**Comment by CasperH2O** [comment]
Hey Super Saiyan 4 Maiko, thanks for reporting. 

Could you double check with the Steam Client open, open Quicktools, navigate to the profile tab and check that the profile you made is now selected? Then check at the bottom of the profile setting to see if the TDP is set there to your wishes.

**Comment by ssj4maiko** [comment]
Yes, I always check that, because I understand that it's based on the active window.

If I'm running a game with TDP to 18W, and move to another window, the Powerplan will be changed, so if it suddenly goes down to 5W, the game will obviously stutter in the background.

Also, some applications (like Steam), may end up running multiple executables, such as the usual "Steam" (_steam.exe_), or the "Gamepad UI" (_\bin\cef.win7x64\steamwebhelper.exe_) as I'm running Beta.

So I mention all these obvious things to explain that I'm aware of it, and I always double check in these cases. (I use Quicktools to make my profiles).

-----------

So most of the time, everything works fine and the profiles are all right, but a few times, as I would open Quicktools, usually on another game, and while changing windows, when I change focus to Steam, I notice that the TDP settings is either ON when it should be off, or high when it should have been set low.

I do remember, however, when I first started using it (2 weeks ago), I would delete all preset configurations (but default, which can't be deleted), yet,  I would find all of them back again later, I tried it a few times until I gave up and tried only to disable or set the TDP down.

So I wonder if this strange revival behavior could be getting mixed up like, overwriting a profile that already exists with a default settings one?

I also do recognize the values 15W and 18W as values that I ended up setting up with "the last game I was configuring", however, when I try to replicate it just like that, it works fine, so there is some other condition causing this problem.

So the reason I'm reporting this bug like this is because I wondered if you maybe would have some idea, like a known bug, or maybe an expected behavior, such as the deleted profiles returning with default options.

If not, I will try to check better next times these happen to see if I find the trigger and report it back

**Comment by CasperH2O** [comment]
Very strange indeed, can't really explain it. Some more questions and possible things to check:

> Also, some applications (like Steam), may end up running multiple executables, such as the usual "Steam" (steam.exe), or the "Gamepad UI" (\bin\cef.win7x64\steamwebhelper.exe) as I'm running Beta.

In that case, have you made multiple profiles with your desired TDP? That might be necessary if Steam switches different applications around.

Are you using the latest release (0.13)? There was indeed a TDP apply bug for active process in 0.12.

**Comment by ssj4maiko** [comment]
> In that case, have you made multiple profiles with your desired TDP? That might be necessary if Steam switches different applications around.

Yes, I made sure both would have either disabled or at low settings (Since deleting the profile won't work anyway).

I say that this bug seems to be mostly happening with Steam, but that's most likely because that's the application that I switch to the most, mostly to test the gamepad and so on. I think it's also not a matter specific to TDP, but to the whole profile, as in, I started disabling some Gyro options, and although I didn't pay much attention, but they may also be reset.

In fact, I think the reason it started resetting to 18W is because I was changing Ryuujinx's profile, which although unrelated to Steam, was the last one that I had changed.

I do have some problems with the gamepad service, but those are most likely related to sleeping the handheld a bit constantly. Usually it works, but sometimes when it's not working and I check the service, it has stopped, a couple times I turned it back on but then there are 2 gamepads recognized, so likely the service being duplicated or something similar.

Of course, turning it off and on again solves everything in that matter. :)

> Are you using the latest release (0.13)? There was indeed a TDP apply bug for active process in 0.12.

0.13.0.1, the auto updater caught on the update, and that's working.

> Are you using any other software that can possibly control TDP like Ayaspace, Power Control Panel or Motion Assistant?

No, although I do have PCP and Motion Assistant installed, I ended up replacing them with your HC, as it had the most practical profile control (PCP's would work most of the time), not going to say it's perfect, as sometimes it crashes during wake up event. But overall, it has been very stable and satisfying, so thank you for the hard work.

**Comment by CasperH2O** [comment]
@ssj4maiko how do you check that TDP is applied? Do you run RTSS/HWinfo64 for example?

Could you provide a step by step with what applications (and their versions) you do, if it's specifically with Steam I can try to reproduce it.

**Comment by ssj4maiko** [comment]
I don't really have any monitor (didn't knew there was one), it's usually between 5 or 15, so I know by the fans and how hot it is (the difference is significant), so when I notice that it's running hotter when it isn't supposed to, I then open the quick menu and confirm that the TDP is wrong, so I readjust it accordingly.

I suspect one way of trying to replicate it would be by alt+tabbing between applications with profiles.

I recently got the opposite, and when I was running Ryuujinx, I noticed it was running slower than usual, so I opened the quick menu, and it had been lowered to 5W, and indeed, I had been alt+tabbing a bit

**Comment by CasperH2O** [comment]
@ssj4maiko respectfully, while I don't doubt your experience and knowledge of your own device, I would feel better if you could do a test with a tool that measures the TDP.  Are you familiar with setting up RTSS and Hwinfo64?

**Comment by ssj4maiko** [comment]
> Are you familiar with setting up RTSS and Hwinfo64?

Nops. Will look into it.

----

I will also update that, I'm not sure if it could be a hint, but I have been getting similar problems again, but in another way (now that I have profiles for most applications I'm using), and what is happening is them suddenly going down to 5W.

One thing of note is that I'm running games and apps at Full Screen (real, not borderless), so when I try to open the quick menu well, as expected, things don't work that well. So I went to play a game that I had played fine, noticed it was running slow, so when Quick Tools bugged out, I would go straight to the main app > Profile > select the profile, and sure enough, the TDP had been changed.

Went to test a game on an emulator (_PS3, which of course, could be slow by default, but I was just testing to see if it was going to work_), and when I noticed the intro movie lagging down, I opened quick tools, and again, 5W.

-----------

Bit I will see the 2 tools you suggested, I was curious about some application that could give me accurate real time data on TDP, considering how much stuff I have on mine

**Comment by CasperH2O** [comment]
@ssj4maiko assuming you are currently running version 0.13, there have been some changes and improvements to the active process detection in development for 0.14. You might want to wait a bit until that's out (no eta, could be a few more weeks). We do have a public beta ongoing on Discord.

You can find it here, not however, that you should only test is if you want for a little bit, don't use it as your main application just yet. There are known issues and changes. You might need to delete all your hotkeys to get it to start. https://drive.google.com/drive/folders/1t39nUXZnY-zq4ees9xXr6BVh5l3tkocX

For HWInfo64 TDP, you'll want to enable OSD for "CPU Package Power".

**Comment by CasperH2O** [comment]
@ssj4maiko are you able to reproduce this issue with HWInfo on and the latest release (0.14)?
